Ct cache fix

elfringham opened the pull request:
Fixes https://github.com/tensorflow/tensorflow/issues/58472 by increasing the element size used to calculate the size of cache needed by the test.

elfringham committed:
Update the cross-trainer cache size in unit test

The calculated number of elements in the dataset is no longer
big enough, so increase the multiplier in the calculation.
This fixes the testConcurrentReaders_test_mode_graph_tfapiversion_2
test.

yangustc07 reviewed (commented):
Could you update the commit description to be more concrete? E.g. Update the cross-trainer cache size in unit test. Thank you.

alankelly committed:
TFLite flex subgraphs reads inputs directly from TF tensors.

PiperOrigin-RevId: 487277120

elfringham commented:
There you are.

tensorflower-gardener committed:
Add int32 support to Sign op

PiperOrigin-RevId: 487282964

jurahul committed:
[XLA] Normalize tuple sharding in FixupPassThroughDomainLinks
- Add a utility to normalize tuple shardings.
- Use it to normalize shardings for tuples in FixupPassThroughDomainLinks

PiperOrigin-RevId: 487283088

tensorflower-gardener committed:
Adding xsymbol fingerprint field to HloModule.

PiperOrigin-RevId: 487284237

ddunl committed:
* Remove uses of tf_cc_test in favor of tsl_cc_test, remove use of core:protos_cc_all in TSL in favor of granular dependencies on specific protos
* Add if_not_android bazel macro to TSL

PiperOrigin-RevId: 487284246

pkanwar23 committed:
upgrade dockerfile to cuda 11.8

PiperOrigin-RevId: 487287054

aaudiber committed:
#tf-data-service Return early on error status.

PiperOrigin-RevId: 487289151

tensorflower-gardener committed:
[XLA/CPU] Reorganize namespaces to eliminate most of the uses of scope
 resolution operator.
- Create impl functions for most of the runtime interface function that
  can reference things without the full scope resolution operator.
- extern "C" interface functions now just call the impl functions.
- eliminate some unused headers

PiperOrigin-RevId: 487290910

tensorflower-gardener committed:
Add int32 support to Abs op

PiperOrigin-RevId: 487295937

faizan-m committed:
Use real FunctionType in FunctionCache (2nd try)

PiperOrigin-RevId: 487300709

cota committed:
xla-opt: register dependent dialects for --hlo-xla-runtime-pipeline

PiperOrigin-RevId: 487301527

tensorflower-gardener committed:
Add JAX as a runtime dependency of TF in preparation for jax2tf support.

PiperOrigin-RevId: 487306476

tensorflower-gardener committed:
Implemented HloRunnerPjRt::ExecuteReplicated() as part of the effort to have HloRunner built on top of PjRt.

PiperOrigin-RevId: 487308910

SandSnip3r committed:
[XLA:GPU] Rollback enabling of cublasLt by default. The is because there are serious performance regressions in various Alphabet models.

PiperOrigin-RevId: 487309888

chhe7 committed:
Supports non-tfl dialect constant ops in fold-constants-to-subgraph pass, because TFL::ConstOp is folded to arith::ConstOp in the canonicalization pass.

PiperOrigin-RevId: 487311124

sampathweb committed:
Removing duplicate section of `tf.keras` in release doc

PiperOrigin-RevId: 487311963

tensorflower-gardener committed:
Remove test_only flags for the hlo runner so that we can use it in prod.

PiperOrigin-RevId: 487312100

jurahul committed:
[XLA/CPU] Reorganize namespaces to eliminate most of the uses of scope
 resolution operator.
- Create impl functions for most of the runtime interface function that
  can reference things without the full scope resolution operator.
- extern "C" interface functions now just call the impl functions.
- eliminate some unused headers

PiperOrigin-RevId: 487316078

d0k committed:
[xla:cpu:next] Add lowerings for infeed and outfeed

This bufferizes mhlo.infeed to lmhlo.infeed and then converts that to XLA
runtime calls. Those forward to the existing runtime functions.

PiperOrigin-RevId: 487317249

Commit:
Change how control dependencies are handled for TPUEmbedding enqueue ops and fix a couple of bugs as well.

PiperOrigin-RevId: 487320020

tensorflower-gardener committed:
Opensource SplitTupleToTensors and InterleaveTensorsToTuple Ops

PiperOrigin-RevId: 487320047

Commit:
[XLA] Move collective_permute_motion to third_party to make it usable by GPUs.

PiperOrigin-RevId: 487320667

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.

PiperOrigin-RevId: 487323063

swachhandl committed:
Refactor out lowering to HLO from XlaCompilationCache into it's own class

Currently the TfGraphToHloCompiler class just calls functions from the XlaCompiler class to do the lowering. Unit tests for this class would be added in the future in case more logic is added there or the lowering logic is moved there instead.

The TfToHloCompiler class is being used as an interface which the TfGraphToHloCompiler class implements.

PiperOrigin-RevId: 487332718

tensorflower-gardener committed:
Change how control dependencies are handled for TPUEmbedding enqueue ops and fix a couple of bugs as well.

PiperOrigin-RevId: 487334153

BlaziusMaximus committed:
Update SavedModel CLI to better support long arguments

PiperOrigin-RevId: 487340517

hubingallin committed:
Switch default cross-device communication option of MirroredStrategy to CollectiveAllReduce

PiperOrigin-RevId: 487341719

fchollet committed:
Add FeatureSpace utility.

PiperOrigin-RevId: 487344904

ishark committed:
Increase test timeout for xent_op_d9m_test to account for flaky timeouts. The test is failing for last few runs.

The test takes close to 5 minutes to complete, running close to the default moderate timeout.

PiperOrigin-RevId: 487345557

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@e07a7040d9c6

Updates LLVM usage to match
[e07a7040d9c6](https://github.com/llvm/llvm-project/commit/e07a7040d9c6)

PiperOrigin-RevId: 487346319

nfelt committed:
Update tensorboard dependency for 2.11.0 release.

PiperOrigin-RevId: 487346430

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 487347846

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/3c654806a782e31a597890bd9a661b1984fb973c.

PiperOrigin-RevId: 487349639

berkinilbeyi committed:
[XLA] Add private stack support to buffer assignment.

Private stacks allow allocating temporary buffers used by async call bodies in a
separate memory space. Private stacks have two benefits:

1- This feature allows backends to implement SPMD-style async calls, where
   multiple threads execute asynchronously in a shared-memory architecture. The
   private stack memory allocation can be then replicated as many times as there
   are threads participating in the SPMD async call so that different async call
   contexts don't clobber each others' temporaries stored in their private
   stacks.

2- This feature allows better temporal reuse of private stack buffers. Heap
   simulator extends live ranges of all buffers including temporaries used in
   async calls, which means that performing module-level heap simulation will
   not place async call temporaries that have disjoint live ranges from the
   perspective of the asynchronous computation will not reuse the same memory
   for these buffers. This CL performs live range analysis for private stack
   buffers from the perspective of the asynchronous computation so it will
   enable reuse of memory for the private stack buffers.

PiperOrigin-RevId: 487362567

tensorflower-gardener committed:
Switch default cross-device communication option of MirroredStrategy to CollectiveAllReduce

PiperOrigin-RevId: 487365854

sagunb committed:
Fix asan issue with QuantizeAndDequantizeV2/V3/V4/V4Grad shape inference functions.

PiperOrigin-RevId: 487366553

ezhulenev committed:
[xla:gpu:runtime] Add matmul plan cache to cuBLASLt custom call

PiperOrigin-RevId: 487372119

Terranlee committed:
Refactor memory viewer, final CL.

PiperOrigin-RevId: 487376657

sagunb committed:
Add the option to propagate assigned device information into the GraphConstructor.

PiperOrigin-RevId: 487380543

Commit:
[XLA:GPU] Add collective-permute motion pass to the GPU compiler

PiperOrigin-RevId: 487383623

tensorflower-gardener committed:
Create OptimizedFunctionGraph proto for saving AoT result.
This proto is used to hold all the information needed to reconstruct optimized function graph.

PiperOrigin-RevId: 487389109

fergushenderson committed:
Add a comment about the return type of tflite_plugin_delegate_create.

PiperOrigin-RevId: 487389734

tensorflower-gardener committed:
Remove unnecessary tf_mlir_bridge_safe_mode flag

We deprecated all the logic and usage regarding mlir_bridge_safe_mode since it is no longer needed after TF/XLA mlir bridge is rolled out and it is used by default.

If this change breaks some existing tests, check if `tf_mlir_bridge_safe_mode` is explicitly set. If so, try to remove it along with tf_mlir_enable_mlir_bridge=true. The default should give a similar behavior.

PiperOrigin-RevId: 487392267

fergushenderson committed:
Fix subtle portability bug with exports in external_delegate_adaptor.cc.

Previously, the declarations of the API functions here were using
TFL_CAPI_EXPORT, which is a macro defined in c_api_types.h that on
most systems expands to '__attribute__((visibility("default")))', but
which on Windows expands to either '__declspec(dllimport)' or
'__declspec(dllexport)' depending on whether a particular flag macro
'TFL_COMPILE_LIBRARY' is defined.  The idea is that when building
the TF Lite C API DLL, you define that symbol, and then the declarations
annotated with TFL_CAPI_EXPORT will be treated as dll exports, causing
that DLL to export those symbols, but when merely using the DLL, you don't
define that symbol, and then those declarations are treated as dll imports,
allowing code outside the DLL to use symbols defined inside the DLL.

However, the symbols declared in this file are NOT intended to be part of the
TF Lite C API DLL, but are instead part of a DIFFERENT shared library,
containing an external delegate.  When building that library, we want
the TF Lite C API symbols to be imports, not exports, but we want the
two symbols defined by that library to be exports.  So we can't use
the same macro for both TF Lite C API symbols and the symbols defined
by that library.  Instead we define a new macro TFL_EXTERNAL_DELEGATE_EXPORT
whose expansion on Windows is controlled by a new flag macro
'TFL_EXTERNAL_DELEGATE_COMPILE_LIBRARY'.

This change is a necessary BUT NOT SUFFICIENT step towards making this
code work on Windows.  (There are definitely other issues, e.g. the build
rule unconditionally generates a .so file rather than a .dll, etc.)

This change should have no effect on non-Windows systems, since the two
macros both expand to the same thing on non-Windows systems.

PiperOrigin-RevId: 487400613

derekmauro committed:
Replace deprecated functions absl::MakeUnique and absl::make_unique
with std::make_unique

PiperOrigin-RevId: 487400683

Commit:
Internal change

PiperOrigin-RevId: 487407993

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@72f89556413f

Updates LLVM usage to match
[72f89556413f](https://github.com/llvm/llvm-project/commit/72f89556413f)

PiperOrigin-RevId: 487426915

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/0b8eec703df52d94ca683cc962110c81951f5900.

PiperOrigin-RevId: 487428271

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/3c1c34c0c0b60dd1d9071040868bb5681d201981.

PiperOrigin-RevId: 487436846

datenglin committed:
Added tests by using multi-process runner to demonstrate periodic pulling and
restarting tasks.

PiperOrigin-RevId: 487444460

tensorflower-gardener committed:
Update GraphDef version to 1311.

PiperOrigin-RevId: 487460515

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2022-11-10

PiperOrigin-RevId: 487460547

akuegel committed:
Add another softmax test with the complete pattern.

PiperOrigin-RevId: 487462191

fergushenderson committed:
Minor doc improvements for the error_reporter_func and error_reporter_user_data fields of TfLiteMinibenchmarkSettings.

PiperOrigin-RevId: 487501677

akuegel committed:
Don't match if the number of elements in the reduce dimension is too small.

The Softmax codegen is only faster than the current emitter if the reduce
dimension is not that large compared to the other dimensions.
Also don't match reductions with type F16, we lower reductions using
gpu.shuffle op which requires F32.

PiperOrigin-RevId: 487519598

jreiffers committed:
Move XLA Framework code from compiler/mlir to compiler/xla.

PiperOrigin-RevId: 487522997

tensorflower-gardener committed:
Automated visibility attribute cleanup.

PiperOrigin-RevId: 487529921

tensorflower-gardener committed:
Merge pull request #58473 from elfringham:ct_cache_fix

PiperOrigin-RevId: 487546017
